src/v/raft/consensus.cc

```diff
@@ -103,14 +103,16 @@ void consensus::maybe_step_down() {
     (void)ss::with_gate(_bg, [this] {
         return _op_lock.with([this] {
             if (_vstate == vote_state::leader) {
-                auto majority_hbeat = config().quorum_match(
-                  [this](model::node_id id) {
-                      if (id == _self) {
-                          return clock_type::now();
-                      }
+                auto majority_hbeat = config().quorum_match([this](vnode rni) {
+                    if (rni == _self) {
+                        return clock_type::now();
+                    }
 
-                      return _fstats.get(id).last_hbeat_timestamp;
-                  });
+                    if (auto it = _fstats.find(rni); it != _fstats.end()) {
+                        return it->second.last_hbeat_timestamp;
+                    }
+                    return clock_type::now();
```

Comment:
It isn't critical but it's better to return something like `clock_type::now() - 10*_jit.base_duration()` instead of `now()` when we can't find a node in _fstats or to avoid the purged follower_stats problem altogether.